tests/configs/config_ds001971.py

```diff
@@ -14,15 +14,11 @@
 
 study_name = 'ds001971'
 task = 'AudioCueWalkingStudy'
-kind = 'eeg'
-plot = False
-reject = {'eeg': 150e-6}
-conditions = ['left', 'right']
-decoding_conditions = [('left', 'right')]
+sessions = [None]
```

Comment:
why is this a list? shouldn't this work without specifying this?

Reply:
no, we expect an iterable at this stage ... in the main config, a simple None would have worked, because then we have something like `if None, make a list of None` --> but we are overwriting the value, so we don't get that control

Reply:
I guess then my question is -> why do you need to rewrite that value? Is there a difference between `None` and `[None]` ?

Reply:
code example:

```Python
In [2]: for i in None: 
   ...:     print('yes') 
   ...:                                                                         
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-2-4625ffe8f4a2> in <module>
----> 1 for i in None:
      2     print('yes')
      3 

TypeError: 'NoneType' object is not iterable

```

We need it to be an iterable, because we are using it as such:

```Python
parallel(run_func(subject, run, session) for subject, run, session in
         itertools.product(config.subjects_list, config.runs, config.sessions))
```

Reply:
but I thought we were using the new updates in `mne-bids` to get the list of sessions of it is `None`. So you should be able to iterate

Reply:
yes, this is how `sessions` gets defined in CONFIG https://github.com/mne-tools/mne-study-template/blob/a623ae84f97baede32a22c9814fa771237a8822a/config.py#L45

so we have a `sessions` variable in CONFIG. However, when we **overwrite** the sessions variable, we are NOT running `get_values_for_key` again ... we are simply overwriting `sessions` - hence, we need an iterable:

https://github.com/mne-tools/mne-study-template/blob/a623ae84f97baede32a22c9814fa771237a8822a/config.py#L650-L660

Reply:
yes that's fine, but I'm still confused :-) why do you need to overwrite when the default is `None`?

Reply:
ahhh got it! see https://github.com/mne-tools/mne-study-template/pull/41/commits/c3123e1be530fe208c08793e02ac3a3de70ff2f7

Reply:
great :-) now you need to make Circle happy ...

Reply:
okay, no ... it fails now. `get_values_for_key` does not return `[None]`, but rather `[]` when no values are found for a key (I think this makes sense).
```Python
    accepted_keys = ('sub', 'ses', 'task', 'run', 'acq')
    if key not in accepted_keys:
        raise ValueError('`key` must be one of "{}". Got "{}"'
                         .format(accepted_keys, key))

    p = re.compile(r'{}-(.*?)_'.format(key))
    value_list = list()
    for filename in Path(bids_root).rglob('*{}-*_*'.format(key)):
        match = p.search(filename.stem)
        value = match.group(1)
        if value not in value_list:
            value_list.append(value)
    return sorted(value_list)
```

Reply:
something needs to be fixed here. Sleep over it ... :)